### PR TITLE
RakeArguments helper

### PR DESCRIPTION
**Pull request by kotp**
This has the advantage of drying the code and makes using it easier.

### CODE REVIEW

File: lib/tasks/exercise_test_task.rb [hunk 1]
```diff
@@ -47,6 +47,6 @@ def run_exercise_tests_in(dir)
   def test_options
     return '' unless ARGV.include?('--')
 
-    ARGV.drop_while { |e| e != '--' }.drop(1).join(' ')
+    Rake.arguments.join(' ')
```
Comment (Insti) [inline lib/tasks/exercise_test_task.rb:50]: This does not do the same thing as the code you're replacing :confused:

File: Rakefile [hunk 2]
```diff
@@ -1,13 +1,14 @@
 require 'rake'
 require 'rake/testtask'
 
+require_relative 'lib/rake_helper'
```
Comment (Insti) [inline Rakefile:4]: Does this need to be in a separate file? Is anything else using it?

File: lib/tasks/exercise_test_task.rb (hunk 1 above)
Comment (kotp) [reply lib/tasks/exercise_test_task.rb:50]: How does this not do so, when the method is exactly the code that was replaced? It passes the `rake test:dev` tests, so is this not tested?

File: Rakefile (hunk 2 above)
Comment (kotp) [reply Rakefile:4]: Yes, `lib/tasks/exercise_test_task.rb` is also using it.

File: lib/tasks/exercise_test_task.rb (hunk 1 above)
Comment (Insti) [reply lib/tasks/exercise_test_task.rb:50]: 'drop_while' != 'take_while'

**Comment by kotp** [comment]
I think this is no longer a valid PR @tommyschaefer but the idea does fit still, I think.

**Comment by tommyschaefer** [comment]
Agreed :+1: I will work naming / removing duplication of argument handling

File: lib/tasks/exercise_test_task.rb (hunk 1 above)
Comment (kotp) [reply lib/tasks/exercise_test_task.rb:50]: Wow, yeah, I missed that... relying on the tests I thought were there.  And the manual run at that time working.

Thanks @Insti

Comment (Insti) [reply lib/tasks/exercise_test_task.rb:50]: Never trust the tests until you've seen them fail.

Comment (kotp) [reply lib/tasks/exercise_test_task.rb:50]: Even then, be slightly suspicious. :)

**Pull request closed without merging**